person.rb

```diff
@@ -19,3 +33,29 @@ def of_age?
     @age >= 18
   end
 end
+
+# Base Decorator
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+# CapitalizeDecorator
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    super().capitalize
+  end
+end
+
+# TrimmerDecorator
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    super()[0..9]
+  end
+end
```

Comment:
- [ ] When trying the testing your code as per the requirements I had the following errors, Kindly try to fix your classes:



<img width="659" alt="Screenshot 2023-11-09 at 2 33 46 PM" src="https://github.com/egichuhis/school-library-ruby/assets/53347966/ce4372ae-dea8-4d0f-b16d-06f8e5ff0bbe">

Reply:
Thanks @algerina for noting that